scipy/optimize/optimize.py

```diff
@@ -865,6 +865,7 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
                                 sim[j] = np.clip(
                                     sim[j], lower_bound, upper_bound)
                             fsim[j] = func(sim[j])
+            iterations += 1
```

Comment:
The issue you fixed here clearly indicate there is a proper test missing. We should not rely on doctests for testing. A bit out of scope of this PR but could still be nice to add.